Rubberduck.Parsing/ComReflection/DeclarationsFromComProjectLoader.cs

```diff
@@ -0,0 +1,307 @@
+﻿using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.InteropServices.ComTypes;
+using Rubberduck.Parsing.Symbols;
+using Rubberduck.VBEditor;
+
+namespace Rubberduck.Parsing.ComReflection
+{
+    public class DeclarationsFromComProjectLoader : IDeclarationsFromComProjectLoader
+    {
+        public IReadOnlyCollection<Declaration> LoadDeclarations(ComProject type, string projectId = null)
+        {
+            var declarations = new List<Declaration>();
+
+            var projectName = new QualifiedModuleName(type.Name, type.Path, type.Name, projectId);
+            var project = new ProjectDeclaration(type, projectName);
+            declarations.Add(project);
+
+            foreach (var alias in type.Aliases.Select(item => new AliasDeclaration(item, project, projectName)))
+            {
+                declarations.Add(alias);
+            }
+
+            foreach (var module in type.Members)
+            {
+                var moduleIdentifier = module.Type == DeclarationType.Enumeration || module.Type == DeclarationType.UserDefinedType
+                    ? $"_{module.Name}"
+                    : module.Name;
+                var moduleName = new QualifiedModuleName(type.Name, type.Path, moduleIdentifier);
+
+                var moduleDeclarations = GetDeclarationsForModule(module, moduleName, project);
+                declarations.AddRange(moduleDeclarations);
+            }
+
+            return declarations;
+        }
+
+        private static ICollection<Declaration> GetDeclarationsForModule(IComType module, QualifiedModuleName moduleName, ProjectDeclaration project)
+        {
+            var declarations = new List<Declaration>();
+
+            var attributes = GetModuleAttributes(module);
+            var moduleDeclaration = CreateModuleDeclaration(module, moduleName, project, attributes);
+            declarations.Add(moduleDeclaration);
+
+            switch (module)
+            {
+                case IComTypeWithMembers membered:
+                    var (memberDeclarations, defaultMember) =
+                        GetDeclarationsForProperties(membered.Properties, moduleName, moduleDeclaration);
+                    declarations.AddRange(memberDeclarations);
+                    AssignDefaultMember(moduleDeclaration, defaultMember);
+
+                    (memberDeclarations, defaultMember) = GetDeclarationsForMembers(membered.Members, moduleName,
+                        moduleDeclaration, membered.DefaultMember);
+                    declarations.AddRange(memberDeclarations);
+                    AssignDefaultMember(moduleDeclaration, defaultMember);
+
+                    if (membered is ComCoClass coClass)
+                    {
+                        (memberDeclarations, defaultMember) = GetDeclarationsForMembers(coClass.SourceMembers,
+                            moduleName, moduleDeclaration, coClass.DefaultMember, true);
```

Comment:
can I haz named argument for the `true` literal here?

Reply:
Will add it